tests/pipelines/test_election.py

```diff
@@ -199,3 +199,27 @@ def test_full_pipeline(b, user_pk):
     tx_from_block = set([tx.id for tx in invalid_block.transactions])
     tx_from_backlog = set([tx['id'] for tx in list(query.get_stale_transactions(b.connection, 0))])
     assert tx_from_block == tx_from_backlog
+
+
+def test_handle_block_events():
+    from bigchaindb.events import setup_events_queue, EventTypes
+
+    events_queue = setup_events_queue()
+    e = election.Election(events_queue=events_queue)
+    block_id = 'a' * 64
+
+    assert events_queue.qsize() == 0
+
+    # no event should be emited in case a block is undecided
+    e.handle_block_events({'status': Bigchain.BLOCK_UNDECIDED}, block_id)
+    assert events_queue.qsize() == 0
+
+    # put an invalid block event in the queue
+    e.handle_block_events({'status': Bigchain.BLOCK_INVALID}, block_id)
+    event = e.event_handler.get_event()
+    assert event.type == EventTypes.BLOCK_INVALID
+
+    # put an valid block event in the queue
+    e.handle_block_events({'status': Bigchain.BLOCK_VALID}, block_id)
+    event = e.event_handler.get_event()
+    assert event.type == EventTypes.BLOCK_VALID
```

Comment:
We can treat this in a separate issue, as a clean up task, but I'll take the opportunity to share some thoughts on one possible approach regarding the imports in tests. There's this idea of not importing a module-under-tests (MUT) at module scope. One helpful reference for this approach is at http://pylonsproject.org/community-unit-testing-guidelines.html

The module-under-test (MUT) can be a function, class, or method for instance. In the test above one could say that the MUT is the class `Election`. As per the recommendation of the approach being discussed here, one would then import the `Election` class under the test testing it. The reason that is given for doing so is:

> Import failures in the module-under-test (MUT) should cause individual test cases to fail, and they should never prevent those tests from being run. Depending on the test runner, import problems may be much harder to distinguish at a glance than normal test failures.

So that's it. We do not need to be maniacal about it though :smile: as this is most likely debatable.